### DISCUSSION TITLE
Sourcerect values below 10 no longer supported

### DISCUSSION THREAD

**Discussion by UnhappyWithMyUsername** [FakeFishGames/Barotrauma#14915] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

As of the Unto the Breach update, the game no longer supports sourcerect X and Y scale values of below "10".
For example the following sourcerect will cause the item to be invisible and no longer show up in debug view nor hightlight menu, effectively rendering it noninteractable and invisible:
`<Sprite texture="%ModDir:3217556378%/Content/Items/II_ItemsAtlas.png" sourcerect="0,53,80,9" depth="0.55" />`
changing the sourcerect to sourcerect="0,53,80,10" makes the item function normally again.

Tested with the "cigarette" item from Immersive Ignitables. I've updated the workshop version to the new sourcerect values for now so people can play without issues. Attached is the old version with the now unsupported sourcerect for you to test with:
[ImmersiveIgnitables.zip](https://github.com/user-attachments/files/17492986/ImmersiveIgnitables.zip)

### Reproduction steps

1. Define a sprite sourcerect and have the X or Y value for scale be below "10"
2. Spawn the item in your inventory. When holding it, it will be invisible
3. Drop the item. You can't see it, not even in debug and can't interact with it via the highlight menu

Alternatively:
1. Extract the attached Immersive Ignitables version into your localmods folder
2. Load the mod and spawn a "Cigarette" in your inventory
3. The cigarette will be invisible and can't be interacted with nor located once dropped.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.6.17.0 (Unto the Breach Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! I think this isn't directly related to the sourcerect: there's an optimization in the latest update that hides item sprites that  have been scaled down to less than one pixel. Considering the scale and tiny sprite of this item, I suspect that's what's happening here. But this does seem like a bug regardless, if the item would be possible to see despite it's small size, it should be rendered. So we'll take a closer look at this, sorry for the inconvenience!

**Comment by Regalis11** [comment]
Fixed in our private repo now (finally), fix coming up in the next update.
